addons/pos_razorpay/static/src/app/screens/payment_screen/payment_screen.js

```diff
@@ -23,4 +24,36 @@ patch(PaymentScreen.prototype, {
             }
         });
     },
+    addNewPaymentLine(paymentMethod) {
+        if (paymentMethod.use_payment_terminal === "razorpay" && this.isRefundOrder) {
+            const refundedOrder = this.currentOrder.lines[0]?.refunded_orderline_id?.order_id;
+            const razorpayPaymentlines = refundedOrder.payment_ids.filter(
+                (pi) => pi.payment_method_id.use_payment_terminal === "razorpay"
+            );
+            const res = super.addNewPaymentLine(paymentMethod);
```

Comment:
```suggestion
            const res = await super.addNewPaymentLine(paymentMethod);
```

the super call returns a promise.